app/addons/documents/doc-editor/components.react.jsx

```diff
@@ -218,44 +218,137 @@ var AttachmentsPanelButton = React.createClass({
+    var attachments = this.props.doc.get('_attachments');
 
-    return _.map(this.props.doc.get('_attachments'), function (item, filename) {
+    function create(item, filename) {
       var url = FauxtonAPI.urls('document', 'attachment', db, doc, app.utils.safeURLName(filename));
       return (
-        <li key={filename}>
-          <a href={url} target="_blank" data-bypass="true"> <strong>{filename}</strong>
-            <span className="attachment-delimiter">-</span>
-            <span>{item.content_type}, {Helpers.formatSize(item.length)}</span>
-          </a>
-        </li>
+          <li key={filename} className="attachment-row">
+            <a href={url} target="_blank" data-bypass="true"> <strong>{filename}</strong>
+              <span className="attachment-delimiter">-</span>
+              <span>{item.content_type}, {Helpers.formatSize(item.length)}</span>
+            </a>
+          </li>
       );
-    });
+    }
+
+    function filter(predicate) {
+      return _.reduce(attachments, function(list, item, filename) {
+        if (predicate(filename)) {
+          list.push(create(item, filename));
+        }
+
+        return list;
+      }, []);
+    }
+
+    if (this.state.filter === '') {
+      return _.map(attachments, create);
+    } else {
+      var strong = new RegExp('^' + this.state.filter);
+      var weak = new RegExp(this.state.filter);
+      return filter(f => strong.test(f)).concat(filter(f => !strong.test(f) && weak.test(f)));
+    }
   },
 
   render: function () {
     if (this.props.isLoading || !this.props.doc.get('_attachments')) {
       return false;
     }
 
+    var ulStyle = {
+      maxHeight: this.state ? this.state.maxHeight : "none"
+    };
+
     return (
-      <div className="panel-section view-attachments-section btn-group">
+      <div className="panel-section view-attachments-section btn-group" ref={this.updateMaxHeight}>
         <button className="panel-button dropdown-toggle btn" data-bypass="true" data-toggle="dropdown" title="View Attachments"
           id="view-attachments-menu">
           <i className="icon icon-paper-clip"></i>
           <span className="button-text">View Attachments</span>
           <span className="caret"></span>
         </button>
-        <ul className="dropdown-menu" role="menu" aria-labelledby="view-attachments-menu">
-          {this.getAttachmentList()}
-        </ul>
+          <ul style={ulStyle} className="dropdown-menu" role="menu" aria-labelledby="view-attachments-menu">
+            <li><AttachmentsPanelFilter /></li>
+            {this.getAttachmentList()}
+          </ul>
       </div>
     );
   }
 });
 
+var AttachmentsPanelFilter = React.createClass({
+  propTypes: {
+    filter: React.PropTypes.string
+  },
+
+  getDefaultProps: function () {
+    return {
+      filter: this._filter
+    };
+  },
+
+  onChange: function (event) {
+    Stores.attachmentFilterStore.updateFilter(event.target.value);
+  },
+
+  stopClick: function (el) {
```

Comment:
This is not the ideal solution. Could you rather add a onClick event to the `dropdown-menu` component and if its the filter it doesn't close it otherwise if its a link it opens it up in another tab. Even better would be to use the dropdown from [React-bootstrap](https://react-bootstrap.github.io/components.html#btn-dropdowns). It has an example with a filter. You don't need to make it as complicated as there example. But you just don't need to do any of these jquery event overrides.